### DISCUSSION TITLE
How to Apply Query LocalScope before Displaying a table?

### DISCUSSION THREAD

**Discussion by jigujigu** [filamentphp/filament#3804] [Help]
Hi All,
 
How I can apply query localscope, before it displaying a table ?

Thank You.

**Comment by danharrin** [comment]
In an admin panel resource?

**Comment by jigujigu** [comment]
Yes.

I mean before records displaying here
![Countries_List](https://user-images.githubusercontent.com/6214964/188081563-81e7d07e-80c5-412f-a417-563b4c20035a.jpg)
Let say I want only showing for ID > 10000.
How i can achieved that ? 
where and what function should I modify ?
Thank You

**Reply by danharrin** [reply]
Override the `getTableQuery()` on the List page and extend `parent::getTableQuery()`

**Reply by jigujigu** [reply]
Hi danharrin,

it gives me error result 'not compatible with method '.
Can you encourage more. please.

Thank you.


**\app\Filament\Resources\CountryResource\Pages\ListCountries.php**
```
<?php

namespace App\Filament\Resources\CountryResource\Pages;

use App\Filament\Resources\CountryResource;
use Filament\Pages\Actions;
use Filament\Resources\Pages\ListRecords;

class ListCountries extends ListRecords
{
    protected static string $resource = CountryResource::class;

    protected function getActions(): array
    {
        return [
            Actions\CreateAction::make(),
        ];
    }

    protected function getTableQuery(): Builder
    {
        return static::getResource()::getEloquentQuery()->where('Id','>=',10000);
    }

}
```


**Error message :**
```
<?php
protected function getTableQuery(): Builder
App\Filament\Resources\CountryResource\Pages\ListCountries::getTableQuery

<?php
protected function getTableQuery(): Builder { }
@return \App\Filament\Resources\CountryResource\Pages\Builder

Method 'App\Filament\Resources\CountryResource\Pages\ListCountries::getTableQuery()' is not compatible with method 'Filament\Resources\Pages\ListRecords::getTableQuery()'.intelephense(1038)
```

**Reply by danharrin** [reply]
`return parent::getTableQuery()->where('Id','>=',10000);` and import the eloquent Builder class
